app/components/pages/PlatformAppMainPage.vue.ts

```diff
@@ -0,0 +1,23 @@
+import Vue from 'vue';
+import { Component, Prop } from 'vue-property-decorator';
+import PlatformAppPageView from '../PlatformAppPageView.vue';
+import { PlatformAppsService, EAppPageSlot } from 'services/platform-apps';
+import { Inject } from 'util/injector';
+
+@Component({
+  components: { PlatformAppPageView },
+})
+export default class PlatformAppMainPage extends Vue {
+  @Prop() params: { appId: string };
+  @Inject() platformAppsService: PlatformAppsService;
+
+  get pageSlot() {
+    return EAppPageSlot.TopNav;
+  }
+
+  get poppedOut() {
+    return !!this.platformAppsService
+      .getApp(this.params.appId)
```

Comment:
Couldn't this fail, even if we control `params.appId`, for instance, when an app gets removed potentially?